cryptoki/src/session/object_management.rs

```diff
@@ -14,6 +14,42 @@ const MAX_OBJECT_COUNT: usize = 10;
 
 impl Session {
     /// Search for session objects matching a template
+    ///
+    /// # Arguments
+    /// * `template` - A [Attribute] of search parameters that will be used
+    ///                 to find objects.
+    ///
+    /// # Example
+    ///
+    /// ```rust
+    /// use cryptoki::session::Session;
+    /// use cryptoki::context::Pkcs11;
+    /// use cryptoki::object::{Attribute, AttributeType, CertificateType, ObjectClass, ObjectHandle};
+    ///
+    /// let mut client = Pkcs11::new(env!("PKCS11_LIBRARY_LOCATION")).expect("Failed to load library");
```

Comment:
```
error: environment variable `PKCS11_LIBRARY_LOCATION` not defined at compile time
 --> src/session/object_management.rs:29:30
  |
8 | let mut client = Pkcs11::new(env!("PKCS11_LIBRARY_LOCATION")).expect("Failed to load library");
  |                              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  |
  = help: Use `std::env::var("PKCS11_LIBRARY_LOCATION")` to read the variable at run time
```

Reply:
Just in case it's helpful a similar code is in https://github.com/parallaxsecond/rust-cryptoki/blob/72673c8cfb2285ab86f485845f1ec439111a8b6a/cryptoki/tests/common.rs#L15